src/Hpack/Config.hs

```diff
@@ -652,9 +687,13 @@ readPackageConfig (DecodeOptions programName file mUserDataDir readValue) = runE
   (warnings, value) <- lift . ExceptT $ readValue file
   tell warnings
   config <- decodeValue programName file value
+  let Product (Product _ configCommonOptions) _ = config
+      commonOptionsWithDefaults = defaultCommonOptions <> configCommonOptions
+      configWithDefaultCommonOptions =
+        first (second (const commonOptionsWithDefaults)) config
```

Comment:
Thus feels a little bit dirty, but I don't have any better idea. I think making it more explicit could help (name it clearly after what's going on here, e.g. `configWithDefaultLanguage` etc, don't abstract early).